exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/okhttp/JsonRequestBody.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.okhttp;
+
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import okio.BufferedSink;
+
+public class JsonRequestBody extends RequestBody {
+  private static final MediaType JSON_MEDIA_TYPE = MediaType.parse("application/json");
+
+  private final byte[] preserializedJson;
+
+  /** Creates a new {@link JsonRequestBody}. */
+  public JsonRequestBody(Marshaler marshaler) {
+    preserializedJson = preserializeJson(marshaler);
```

Comment:
Initially I wanted to save off the marshaler like ProtoResponseBody does and substitute writeBinaryTo() with writeJsonTo() in the overridden writeTo() method below. However there wasn't an equivalent method for getting the contentLength like ProtoResponseBody does in its constructor, so this seemed like the simplest way to do this with the existing code at my disposal. 

I copied the implementation from `MarshalerUtil.preserializeJsonFields()` but omit the part that trims the beginning and ending brackets.

Reply:
How about just sticking with `-1` for `contentLength` without preserializing?

Reply:
Interesting. I initially didn't think we could leave the contentLength unset because we do some validation on our end to see that it is a positive value. But I don't see anything in the spec that indicates the `Content-Length` header must be set and I don't see anything in our code that relies on it other than for monitoring. Asking my team for any context around this choice, but so far I'm inclined to do as you suggest and return -1.

Reply:
Though, might I ask, is there a downside to leaving this as is? I _think_ it's just a matter of when this serialization logic gets called. It's a bit wasteful to do it this way as there is an extra copy cost to transfer the bytes into the final buffer, but I don't think by very much.

Reply:
It's not a big deal either way if only for tests - but OkHttp's intended usage is to "do work" on its I/O thread, so it's a bit more correct to do serialization within `writeTo`, whereas preserialization in the constructor is serializing on the hot path in principle (not if it's a test, but otherwise it would be).